mlflow/store/tracking/file_store.py

```diff
@@ -348,17 +355,14 @@ def create_experiment(self, name, artifact_location=None, tags=None):
         self._check_root_dir()
         _validate_experiment_name(name)
         self._validate_experiment_does_not_exist(name)
-        # Get all existing experiments and find the one with largest numerical ID.
-        # len(list_all(..)) would not work when experiments are deleted.
-        experiments_ids = [
-            int(e)
-            for e in (
-                self._get_active_experiments(full_path=False)
-                + self._get_deleted_experiments(full_path=False)
+        experiment_id = _generate_unique_integer_id()
+        if not self._has_experiment(FileStore.DEFAULT_EXPERIMENT_ID):
+            warnings.warn(
+                "The default experiment is not present due to a file system issue or manual "
```

Comment:
If we disallow deleting the default experiment, would that simplify the implementation?

Reply:
We can disallow in the delete experiment API :)